aws_lambda_powertools/utilities/data_classes/kinesis_firehose_event.py

```diff
@@ -1,9 +1,123 @@
 import base64
-from typing import Iterator, Optional
+import json
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, Iterator, List, Optional
+
+from typing_extensions import Literal
 
 from aws_lambda_powertools.utilities.data_classes.common import DictWrapper
 
 
+@dataclass
+class KinesisFirehoseResponseRecordMetadata:
+    """
+    Documentation:
+    --------------
+    - https://docs.aws.amazon.com/firehose/latest/dev/dynamic-partitioning.html
+    """
+
+    partition_keys: Optional[Dict[str, str]]
+
+    def asdict(self) -> Optional[Dict]:
+        if self.partition_keys is not None:
+            return {"partitionKeys": self.partition_keys}
+        return None
+
+
+@dataclass
+class KinesisFirehoseResponseRecord:
+    """Record in Kinesis Data Firehose event
```

Comment:
May want to change this to "Record in Kinesis Data Firehose response object"

Reply:
+1